lib/modules-preprocessor.js

```diff
@@ -15,20 +17,40 @@ module.exports = class ModulesPreprocessor {
   toTree(inputTree, path) {
     if (path !== '/') { return inputTree; }
 
-    let merged = new MergeTrees([inputTree, this.getModulesTree()], { overwrite: true });
-
-    // Exclude the individual CSS files – those will be concatenated into the styles tree later
-    return new Funnel(merged, { exclude: ['**/*.' + this.owner.getFileExtension()] });
+    return new MergeTrees(
+      [
+        new Funnel(inputTree, {
+          exclude: ['**/*.' + this.owner.getFileExtension()]
+        }),
+        this.getModulesTree()
+      ],
+      { overwrite: true }
+    );
   }
 
   getModulesTree() {
     if (!this._modulesTree) {
-      let inputRoot = this.owner.belongsToAddon() ? this.owner.getParentAddonTree() : this.owner.app.trees.app;
-      let outputRoot = (this.owner.belongsToAddon() ? this.owner.getAddonModulesRoot() : '');
+      const isMU = isModuleUnification(this.owner.parent);
+      const belongsToAddon = this.owner.belongsToAddon();
+
+      let inputRoot = belongsToAddon
+        ? this.owner.getParentAddonTree()
+        : isMU
+          ? this.owner.app.trees.src
+          : this.owner.app.trees.app;
+
+      let outputRoot = isMU
+        ? path.join(
+            this.owner.getOwnerName(),
+            (belongsToAddon ? this.owner.parent : this.owner).treePaths.src
```

Comment:
I don't think we ever want to reference `this.owner.treePaths` (since `owner` is an instance of the ECM addon itself)—maybe this should be something like `this.owner[belongsToAddon ? 'parent' : 'app'].treePaths.src`?

Reply:
Actually, fun story: I just totally unrelatedly discovered that `EmberApp` doesn't have `treePaths`